pages/_app.js

```diff
@@ -15,6 +16,36 @@ export default function App({ Component, pageProps }) {
     setEntriesList([newEntry, ...entriesList]);
   }
 
+  const [images, setImages] = useState(pictures);
+  const [filter, setFilter] = useState();
+  /* 
+  function handleFavoriteImages() {
+    setFilter("favorites");
+    //console.log(filter);
+  } */
+
+  /* const favoriteImages = images.filter((image) => image.isFavorite === true);
+  console.log(favoriteImages, "fav"); */
+
+  function handleToggleFavorite() {
+    setImages(
+      pictures.map((picture) =>
+        picture.id === "1"
+          ? { ...picture, isFavorite: !picture.isFavorite }
+          : picture
+      )
+    );
```

Comment:
And here you accept the id of an image that you want to toggle.
In addition, I changed the original `pictures` to `images` because you want to rely on the most recent state and not on the initial state.
```suggestion
  function handleToggleFavorite(id) {
    setImages(
      images.map((image) =>
        image.id === id
          ? { ...image, isFavorite: !image.isFavorite }
          : image
      )
    );
```